src/Univ2LpOracle.sol

```diff
@@ -247,7 +249,26 @@ contract UNIV2LPOracle {
 
         // Get LP token supply
         uint256 supply = ERC20Like(src).totalSupply();
-        require(supply != 0, "UNIV2LPOracle/invalid-lp-token-supply");
+        require(supply > 0, "UNIV2LPOracle/invalid-lp-token-supply");
+
+        // Adjust LP token supply if fees are on
+        // We sacrifice some correctness for gas-savings by not querying param `feeTo`
+        // from the factory. This is only an issue during the time interval that Uniswap
+        // LP fees are toggled on/off and liquidity in the pair is added/removed.
+        // Uniswap Governance toggling fees is assumed to be a rare event, with liquidity
+        // for popular pairs constantly in flux. In the worst case the LP price would be off
+        // by ~0.05% which is insigificant relative to drift caused by the `hop` delay.
+        uint256 kLast = UniswapV2PairLike(src).kLast();
+        if (kLast > 0) {
+            uint256 rootK = sqrt(k);
+            uint256 rootKLast = sqrt(kLast);
+            if (rootK > rootKLast) {
+                uint256 numerator = wmul(supply, sub(rootK, rootKLast));
+                uint256 denominator = add(wmul(rootK, 5), rootKLast);
```

Comment:
I don't think you want to use `wmul` here...that will result in `rootK * 5 / 10**18`, but actually you need `rootK * 5` so you should just use `mul`.

Reply:
fixed